### ISSUE TITLE
[story] doesn't work as ActionWML

### ISSUE THREAD

**Issue by Samonella7** [wesnoth/wesnoth#3366]
Using BfW 1.14.4 from Steam, on Windows 10

I came across something weird today. According to the [wiki](https://wiki.wesnoth.org/IntroWML) [story] can be used as action wml now. For example, take a simple scenario containing this:
```
[event]
    name=prestart
    [story]
        [part]
            story=_"Hello"
        [/part]
    [/story]
    [endlevel]
        result=victory
    [/endlevel]
[/event]
```
It's just a campaign's epilogue, displaying the last bits of story before the credits roll. 

When the scenario runs, however, the event is interrupted with this message: 
`<Invalid WML found> Missing "title" key in [story] ActionWML`
Which is very weird; [part]s have an optional "title" key, but by the wiki [story] doesn't even have that. Anyway, I add `title="title"` to the [story] tag and it works. Not sure why it's necessary, but it sounds like a simple wiki error.

But wait, it gets weirder.

Back in my actual campaign, leaving out the title key doesn't make a WML error, I get a fatal error:
```
Assertion failed!

Program C:\Program Files
(x86)\Steam\steamapps\common\wesnoth\wesnoth.exe
File: src/play_controller.cpp, Line 692

Expression: mod > 0
```
I haven't managed to narrow down what made the differnce. I also get the fatal error on a non-steam installation of 1.14.2 (still on Windows 10).

So anyway, attached is a test campaign demonstrating it. Start it up and you get the wml error, then in _main swap out which `first_scenario=` line is commented and you should get the assertion error. And in either scenario, adding the `title` key to [story] makes the event work right.

**Comment by Vultraz** [comment]
If you want story in prestart, you should just place the tag at top-level right under [scenario].

The title thing sounds like an oversight, or perhaps a wiki error like you said. And the assertion looks like something that perhaps happens due to trying to use [story] in prestart... but that's just a guess.

**Comment by Samonella7** [comment]
> If you want story in prestart, you should just place the tag at top-level right under [scenario].
Right, normally I would. I ran into this using [insert_tag] to varying story messages depending on which characters survived, so the [story] had to go after the code that set up the story variable. I guess I could set it up in the previous scenario... but still, the [insert_tag] works if I include the `title` key.

**Comment by CelticMinstrel** [comment]
The title _is_ required for ActionWML story tags (in `[story]`, not `[part]`) if I recall correctly; for scenario-level tags it defaults to the scenario's title. So at least in the first part this is behaving as designed, though there's still the open question of whether the design should be changed. I don't know what's causing the assertion.

**Comment by jostephd** [comment]
The odd thing about `title` is that there's an error if it's missing, but after adding it it doesn't seem to be used. It's not shown anywhere user-visible, for example.

**Comment by CelticMinstrel** [comment]
If I recall correctly, it's used if you have a `[part]` that doesn't have its own `title`. So maybe raising an error when it's missing is a bit much?

**Comment by nemaara** [comment]
Not sure when, but it seems to be fixed now. See this code from DiD, which seems to run without issue [https://github.com/wesnoth/wesnoth/blob/f18e24caf04b37eace6c44157660599387fa4e55/data/campaigns/Descent_Into_Darkness/scenarios/12_Endless_Night.cfg#L1207](url).
